File: LeetCode/Greedy/Golang/Leetcode908-最小差值.go
package sbgo

func smallestRangeI(nums []int, k int) int {
	x, y := 100000, -1
	for _, e := range nums {
		x = min(x, e)
		y = max(y, e)
	}
	return max(0, y-x-2*k)
}

func max(i, j int) int {
	if i < j {
		return j
	}
	return i
}

func min(i, j int) int {
	if i < j {
		return i
	}
	return j
}

File: LeetCode/Greedy/JS/Leetcode908-最小差值.js
var smallestRangeI = function (nums, k) {
    let x = 100000, y = -1
    for (const e of nums) {
        x = Math.min(x, e);
        y = Math.max(y, e);
    }
    return Math.max(0, y - x - 2 * k);
};